### PR TITLE
Add the localization-internationalization to support Ukrainian language

### PR DESCRIPTION
**Pull request by AviGawande**
This PR adds  #148 the Ukrainian Language support to the Eventyay-tickets through Internationalization and Localization,
this pr will be updated by commiting more code updates into the branch

### CODE REVIEW

File: src/pretix/helpers/locale/uk/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: PACKAGE VERSION\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2024-06-18 05:58+0000\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
```
Comment (mariobehling) [inline src/pretix/helpers/locale/uk/LC_MESSAGES/django.po:18]: Not sure why this is here, LL@li.org

File: src/pretix/locale/djangojs.pot
Comment (mariobehling) [inline src/pretix/locale/djangojs.pot:1]: Not sure why this gets deleted.

File: src/pretix/control/forms/global_settings.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ def __init__(self, *args, **kwargs):
             )),
             ('opencagedata_apikey', SecretKeySettingsField(
                 required=False,
-                label=_("OpenCage API key for geocoding"),
+                label=_("OpenCage API key for geo   ding"),
```
Comment (norbusan) [inline src/pretix/control/forms/global_settings.py]: Please revert this change, it is unrelated, and wrong.

File: src/pretix/settings.py [hunk 3]
```diff
@@ -449,12 +449,13 @@
     ('ru', _('Russian')),
     ('es', _('Spanish')),
     ('tr', _('Turkish')),
+    ('uk', _('Ukrainian')),
```
Comment (norbusan) [inline src/pretix/settings.py]: uk is United Kingdom
ua is Ukraine

File: src/pretix/settings.py [hunk 4]
```diff
@@ -449,12 +449,13 @@
...
 ]
 LANGUAGES_OFFICIAL = {
     'en', 'de', 'de-informal'
 }
 LANGUAGES_INCUBATING = {
-    'pl', 'fi', 'pt-br'
+    'pl', 'fi', 'pt-br','uk'
```
Comment (norbusan) [inline src/pretix/settings.py]: As above, it is NOT `uk`

**Review by norbusan — CHANGES_REQUESTED** [review]

File: src/pretix/control/forms/global_settings.py (hunk 2 above)
Comment (AviGawande) [reply src/pretix/control/forms/global_settings.py]: ok

File: src/pretix/settings.py (hunk 4 above)
Comment (AviGawande) [reply src/pretix/settings.py]: sure

File: src/pretix/settings.py (hunk 3 above)
Comment (AviGawande) [reply src/pretix/settings.py]: oh ! i'll correct

**Comment by mariobehling** [comment]
Resolve conflicts.

**Comment by AviGawande** [comment]
i updated the tranlsation PO files for ukrainian language 

> Resolve conflicts.

**Comment by mariobehling** [comment]
Conflicting files

Please resolve conflicts.

**Comment by mariobehling** [comment]
Please do not open several pull request for the same thing. Stick to the PR and fix any issues in the same PR. Thanks.

**Review by mariobehling — CHANGES_REQUESTED** [review]
Screenshot from 2024-06-21 17-16-52

Ukrainian is no-where listed as a language option.

**Comment by mariobehling** [comment]
@norbusan The language code for Ukraine is indeed "uk". For the UK English the language code is en-gb.

Screenshot from 2024-06-21 17-43-34
Screenshot from 2024-06-21 17-41-41

@AviGawande Please change this and implement the functionality including adding Ukrainian to the UI options before the start of the next week.

File: src/pretix/locale/djangojs.pot
Comment (AviGawande) [reply src/pretix/locale/djangojs.pot:1]: yes trying to figure this out as well

**Comment by AviGawande** [comment]
okay, will fix those changes in the current branch.

**Comment by AviGawande** [comment]
> @norbusan The language code for Ukraine is indeed "uk". For the UK English the language code is en-gb.
> 
> Screenshot from 2024-06-21 17-43-34 Screenshot from 2024-06-21 17-41-41
> 
> @AviGawande Please change this and implement the functionality including adding Ukrainian to the UI options before the start of the next week.

**Comment by AviGawande** [comment]
ok @mariobehling earlier when the ukrainian language code when set to = ua it was working perfectly but when i changed the code to uk there is some frontend issue with it posting before both the images.
1. language code was set to ua this is the result
Screenshot (38)
Screenshot (39)

2.when language code is set to uk
Screenshot (36)
Screenshot (37)

**Comment by mariobehling** [comment]
Yes, I hope you understand my earlier comments. "uk" is correct. Let's move forward.

**Comment by AviGawande** [comment]
> Yes, I hope you understand my earlier comments to us "uk". Let's move forward.

yes,okay

**Comment by mariobehling** [comment]
As you can see in the tests the Docker build fails:

[Push Docker image to Docker hub](https://github.com/fossasia/eventyay-tickets/actions/runs/9624347608/job/26547954453#step:4:3318)
buildx call failed with: ERROR: failed to solve: process "/bin/sh -c chmod +x /usr/local/bin/pretix &&     rm /etc/nginx/sites-enabled/default &&     cd /pretix/src &&     rm -f pretix.cfg && \tmkdir -p data &&     chown -R pretixuser:pretixuser /pretix /data data && \tsudo -u pretixuser make production" did not complete successfully: exit code: 2

**Comment by AviGawande** [comment]
i will complete those tests to pass

**Comment by AviGawande** [comment]
@mariobehling there are some problem while commiting the changes in add-ukrain-lang branch after development branch is merged into my local branch

**Comment by AviGawande** [comment]
> @mariobehling there are some problem while commiting the changes in add-ukrain-lang branch after development branch is merged into my local branch

tried fetching and merging the changes from development to local branch but nothing working

**Comment by mariobehling** [comment]
Docker PR Build still failing

File: src/pretix/helpers/locale/uk/LC_MESSAGES/django.po (hunk 1 above)
Comment (hongquan) [reply src/pretix/helpers/locale/uk/LC_MESSAGES/django.po:18]: @mariobehling It is just a placeholder, `gettext` expects team to update it.

File: src/pretix/locale/uk/LC_MESSAGES/django.po [hunk 5]
```diff
@@ -3,97 +3,105 @@
...
-"Language-Team: LANGUAGE <LL@li.org>\n"
+"Last-Translator: AbhishekG <gawandeabhi123@gmail.com>\n"
+"Language-Team: Ukrainian <https://translate.pretix.eu/projects/pretix/pretix/uk/>\n"
+
 "Language: \n"
 "MIME-Version: 1.0\n"
 "Content-Type: text/plain; charset=UTF-8\n"
 "Content-Transfer-Encoding: 8bit\n"
-"Plural-Forms: nplurals=INTEGER; plural=EXPRESSION;\n"
+"Plural-Forms: nplurals=4; plural=(n % 1 == 0 && n % 10 == 1 && n % 100 != "
```
Comment (hongquan) [inline src/pretix/locale/uk/LC_MESSAGES/django.po:21]: How did you generate this code? The error in [Docker build job](https://github.com/fossasia/eventyay-tickets/actions/runs/9624347608/job/26547954453#step:4:3318) is related to this.


> ValueError: invalid token in plural form: EXPRESSION

**Review by hongquan — CHANGES_REQUESTED** [review]

File: src/pretix/locale/uk/LC_MESSAGES/django.po (hunk 5 above)
Comment (hongquan) [reply src/pretix/locale/uk/LC_MESSAGES/django.po:21]: Oh..., maybe the error message is caused by other file, because in this hunk, you are deleting `EXPRESSION`. Gettext will not see it.

**Comment by hongquan** [comment]
Only find this file which has `EXPRESSION`: https://github.com/fossasia/eventyay-tickets/blob/a292acb338eb0c8d30af6dfc7989c0c74da7da5b/src/pretix/locale/nb_NO/LC_MESSAGES/djangojs.po#L19

and it may be the culprit of the build failure.

But don't know why this PR makes this file exposed. If this file is the culprit, the build job should be failed before.

**Comment by hongquan** [comment]
@mariobehling 

Before, on the Eventay Telegram group chat, I heard that we had an issue with Docker building script mounting the old version of the source code, instead of the current one. I don't know if this issue (of Docker build) has been fixed. If it is not fixed, it is  likely that the CI failure of this PR is due to it, because the _src/pretix/helpers/locale/ua/LC_MESSAGES/django.po_ file contained `EXPRESSION` before this PR.

**Comment by mariobehling** [comment]
> Before, on the Eventay Telegram group chat, I heard that we had an issue with Docker building script mounting the old version of the source code, instead of the current one. 

@hongquan This was related to the video component. Here things should be working smoothly.

File: src/pretix/locale/uk/LC_MESSAGES/django.po (hunk 5 above)
Comment (AviGawande) [reply src/pretix/locale/uk/LC_MESSAGES/django.po:21]: this django.po file is generated using = python manage.py makemessages -l uk --->> this files extracts the translation 
and then = python manage.py compilemessages to compile the messages.

File: src/pretix/helpers/locale/uk/LC_MESSAGES/django.po (hunk 1 above)
Comment (AviGawande) [reply src/pretix/helpers/locale/uk/LC_MESSAGES/django.po:18]: and the <LL@li.org> is just the placeholder we can replace it with  real one of developer who is translating the language.
